tests/integration/targets/test_gcs_backend/tasks/main.yml

```
@@ -0,0 +1,121 @@
+---
+- name: Test Role 'gcs_backend'
+  block:
+    # Create GCS backend
+    - name: Create GCS backend for Terraform
+      ansible.builtin.include_role:
+        name: cloud.terraform_ops.gcs_backend
+      vars:
+        gcs_backend_operation: create
+        gcs_backend_bucket_name: "{{ test_gcs_backend_bucket_name }}"
+        gcs_backend_gcp_project: "{{ GCP_PROJECT_ID }}"
+        gcs_backend_auth_kind: serviceaccount
+        gcs_backend_service_account_file: "{{ GCP_SERVICE_ACCOUNT_CREDENTIALS_FILE }}"
+
+    - name: Get storage bucket information
+      ansible.builtin.command: gcloud storage buckets describe gs://"{{ test_gcs_backend_bucket_name }}" --format="json(name)"
+      register: bucket_info
+      ignore_errors: true
```

Why should we ignore errors here?

---

if bucket is not present, the command fails causing playbook to fail at that task
I thought of having a `assert` on the `bucket_info` would provide cleaner failure with `assertion failure` if bucket is not present

---

although I'm fine with either way, would you recommend removing ignore errors and letting playbook fail at gcloud command task?

---

Based on [this link](https://ansible.readthedocs.io/projects/lint/rules/ignore-errors/), it seems that ansible-lint would generate an error for this line. Could we consider using one of the recommended methods mentioned in the link?

---

one of the examples under `correct code` in the link uses `register`, I think the current code should work as it has `register` wherever there is `ignore_errors`
```
---
- name: Example playbook
  hosts: all
  tasks:
    - name: Run apt-get update
      ansible.builtin.command: apt-get update
      ignore_errors: true
      register: ignore_errors_register # <- Stores errors and failures for evaluation.
```

---

@mandar242 please do not use any occurrence of `gcloud` cli anywhere.
You can either create a new info module into the google.cloud collection (I dont if we are authorized to do so), create a local python script or ansible module intended to be used only for this integration test target, or a quick an dirty solution is calling again `google.cloud.gcp_storage_bucket` with the same storage bucket name and ensure the module did not reported any changes
Another possible solution to validate that the Storage bucket has been created is to run a terraform configuration using this storage bucket as configuration and then you can use the `gcp_storage_object` with the `download` option to ensure that the state file has been stored

---

fixed, thanks!

---

I also don't understand why we are ignoring errors here.

---

@hakbailey code that was using `gcloud cli` has been removed. I had forgot to resolve this conversation 😅